### ISSUE TITLE
RBMK operation behaviour and building glitches [1.7.10 Issue (Thanks BallOfEnergy)]

### ISSUE THREAD

**Issue by TheChosenIndividual** [HbmMods/Hbm-s-Nuclear-Tech-GIT#1115]
While building RBMKs for testing i have run into a few glitches regarding the RBMK reactor that i have identified and will explain here:

1. Covers disappearing from on top of RBMK columns: Despite the cover being there, a visual bug (doesnt affect functionality) causes the RBMK cover to dissapear, causing the column to appear uncovered, it isnt limited to one type of cover, both suffer the same effect, it seems to be random and not tied to any specific event

2. Columns not displaying properly: When placing an RBMK column of ANY type there is an instance where the RBMK column will appear of the incorrect height, it does not have anything to do with the dialColumnHeight gamerule, as it remains on 4, again, this glitch is purely visual, but leaves an invisible block-space you can walk on, its usually self resolving but breaks the immersion

3. Americium-242 bugging the reactor: When doing trials with HEA-242 i came across a strange bug where the reactor behaviour changes and thinks its a ReaSim upon inserting the HEA-242 and starting the reaction, while the gamerule in reality remains untouched, it completely messes up the functioning of the reactor, Steam will accumulate instead all the component columns instead of the usual steam column generation, the Americium will also not react normally with oncoming neutrons, this glitch can ONLY be resolved that i know of, by rebuilding the ENTIRE reactor core, it can also crash the game if you attempt to change the gamerule while the glitch is happening, it doesnt happen with any other fuel, just HEA-242, it will also glitch any other rods you load into the reactor without rebuilding the reactor, making them effectively useless, wasting materials

I hope this leads to a prompt solution of these bugs, my only intention is to improve community experience

Thanks for reading

-Chosen

**Comment by TheChosenIndividual** [comment]
Edit: heres a screenshot of issue #2, hope it helps understand what is going on
![image](https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/assets/120053489/21fd4ab3-0d38-4503-8f13-5418a726ea15)

**Comment by BallOfEnergy1** [comment]
This needs the 1.12.2 tag, also I've got no clue why this happens other than the 1.12.2 version being buggy.

**Comment by BallOfEnergy1** [comment]
oh yeah also #1039

**Comment by TheChosenIndividual** [comment]
this isnt a 1.12.2 issue, i encountered it in the 1.7.10 releases

**Comment by Voxelstice** [comment]
do you have a crash report for the third issue and are you using any mods like optifine (it probably shouldn't cause any issues but it's possibly the cause)

**Comment by TheChosenIndividual** [comment]
> do you have a crash report for the third issue and are you using any mods like optifine (it probably shouldn't cause any issues but it's possibly the cause

No, I am using optifine, but it isnt the root of the issue, i have tried turning it off and this glitch still appears, however i do not have any crash reports on the Am-242 situation, i will try to get one if i have time

**Comment by Bencek123** [comment]
i have also encountered the missing cover but i have also gotten some missing control rod i mean the thing that lifts up from the column itself

**Comment by BallOfEnergy1** [comment]
I haven't experienced anything like what is described on 1.7.10, do you have any screenshots of it occurring in 1.7.10?

**Comment by Bencek123** [comment]
> I haven't experienced anything like what is described on 1.7.10, do you have any screenshots of it occurring in 1.7.10?

maybe one of the new updates now fixed it cause i can only see the cover dissapearing bug

**Comment by TheChosenIndividual** [comment]
> > I haven't experienced anything like what is described on 1.7.10, do you have any screenshots of it occurring in 1.7.10?
> 
> maybe one of the new updates now fixed it cause i can only see the cover dissapearing bug

I havent experienced it in a bit, but i have also encountered the missing control rod servo/top, this issue has happened in 1.7.10, all of them have

**Comment by BallOfEnergy1** [comment]
> > > I haven't experienced anything like what is described on 1.7.10, do you have any screenshots of it occurring in 1.7.10?
> > 
> > 
> > maybe one of the new updates now fixed it cause i can only see the cover dissapearing bug
> 
> I havent experienced it in a bit, but i have also encountered the missing control rod servo/top, this issue has happened in 1.7.10, all of them have

I've never in my history of NTM seen any such things happen, do you have any rendering mods that could potentially be conflicting or causing issues with rendering NTM RBMK columns?

**Comment by TheChosenIndividual** [comment]
> > > > I haven't experienced anything like what is described on 1.7.10, do you have any screenshots of it occurring in 1.7.10?
> > > 
> > > 
> > > maybe one of the new updates now fixed it cause i can only see the cover dissapearing bug
> > 
> > 
> > I havent experienced it in a bit, but i have also encountered the missing control rod servo/top, this issue has happened in 1.7.10, all of them have
> 
> I've never in my history of NTM seen any such things happen, do you have any rendering mods that could potentially be conflicting or causing issues with rendering NTM RBMK columns?

Not that im aware of

**Comment by RagexPrince683** [comment]
I have also noticed covers disappearing in 1.7.

**Comment by Kal19801** [comment]
This can be easily triggered by placing a lot (exactly how much depends on the performance of the device) of RBMK components

**Comment by Kal19801** [comment]
Adjusting the height of the RBMK by using dialColumnHeight to make a single rbmk component taller will reduce the number of components needed to trigger this bug
